test/IdentityManager.js

```diff
@@ -42,499 +68,357 @@ contract('IdentityManager', (accounts) => {
+
+      tx = await identityManager.cancelMigration(proxy.address, {from: user1})
+      log = tx.logs[0]
+      assert.equal(log.event, 'MigrationCanceled', 'wrong event initiated')
+      assert.equal(log.args.identity, proxy.address, 'canceled migrating wrong proxy')
+      assert.equal(log.args.newIdManager, newIdenManager.address, 'canceled migration to wrong location')
+      assert.equal(log.args.instigator, user1, 'canceled migrating from wrong user')
+
+      //set up migration again
+      tx = await identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: user1})
+      //Second migration attempt, should allow
+      log = tx.logs[0]
+      assert.equal(log.event, 'MigrationInitiated', 'wrong event initiated')
+      assert.equal(log.args.identity, proxy.address, 'migrating wrong proxy')
+      assert.equal(log.args.newIdManager, newIdenManager.address, 'migrating to wrong location')
+      assert.equal(log.args.instigator, user1, 'started migrating from wrong person')
+
+      await evm_increaseTime(userTimeLock)
+      tx = await identityManager.cancelMigration(proxy.address, {from: user2})
+      //young owner should also be able to cancel migration
+      log = tx.logs[0]
+      assert.equal(log.event, 'MigrationCanceled', 'wrong event initiated')
+      assert.equal(log.args.identity, proxy.address, 'canceled migrating wrong proxy')
+      assert.equal(log.args.newIdManager, newIdenManager.address, 'canceled migration to wrong location')
+      assert.equal(log.args.instigator, user2, 'canceled migrating from wrong person')
+
+      await identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: user1})
+      //Don't need to check setup again
+      let threwError = false
+      try {
+        await identityManager.cancelMigration(proxy.address, {from: nobody})
+      } catch(error) {
         assert.isTrue(error.toString().indexOf("invalid JUMP") != -1, 'Should have thrown, did not')
-      }).then(() => {
-        done()
-      })
+        threwError = true
+      }
+      assert.isTrue(threwError, 'Should have thrown error')
     })
 
-    it('correct keys should finilize transfer', (done) => {
-      identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: user1}).then(tx => {
-        return identityManager.finalizeMigration(proxy.address, {from: nobody})
-      }).then(() => {
-        assert.isFalse(true, 'non-owner should not be able to finalize')
-      }).catch(error => {
+    it('correct keys should finilize transfer', async function() {
+      await identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: user1})
+      let threwError = false
+      try {
+          await identityManager.finalizeMigration(proxy.address, {from: nobody})
+      } catch(error) {
         assert.isTrue(error.toString().indexOf("invalid JUMP") != -1, 'should have thrown, did not')
```

Comment:
A nice thing I stole from Pelle: assert.match(error.message, /invalid JUMP/, "should have thrown an error, did not")